src/lib.rs

```diff
@@ -191,21 +194,32 @@ pub mod pallet {
 		pub fn show_me_the_money(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
 			let payee = ensure_signed(origin)?;
 
-			// Calculate the veted amount on demand.
+			// Calculate the vested amount on demand.
 			let mut info =
 				AccountsPayable::<T>::get(&payee).ok_or(Error::<T>::NoAssociatedClaim)?;
 			ensure!(
 				info.claimed_reward < info.total_reward,
 				Error::<T>::RewardsAlreadyClaimed
 			);
+
+			ensure!(
+				info.total_reward > T::MinimumContribution::get(),
+				Error::<T>::ContributionNotHighEnough
+			);
```

Comment:
If we put the check here, we have to run it (and users have to pay for it) every time they make a payout.

I think it makes more sense to do this check when initializing where we can do it once and it won't contribute to user fees.

I agree with your point that doing it there will give us a less complete record of who contributed, but I would argue that the canonical record of who contributed lives on the relay chain. The only thing this data is needed for on the para side is for reward payouts.